### PR TITLE
Add Prelude/Map/unpackOptionals

### PR DESCRIPTION
**Pull request by JackKelly-Bellroy**
I needed this when implementing https://github.com/regadas/github-actions-dhall/pull/20 and it feels generic enough that it would be useful in the prelude.

I noticed in #1053 all the relative imports are now using `.dhall` extensions; should I edit my PR to import `unpackOptionals.dhall`?

Also, is there a script to recursively re-freeze files when you change the prelude?

### CODE REVIEW

File: Prelude/Map/package.dhall [hunk 1]
```diff
@@ -19,6 +19,9 @@
 , map =
       ./map sha256:23e09b0b9f08649797dfe1ca39755d5e1c7cad2d0944bdd36c7a0bf804bde8d0
     ? ./map
+, unpackOptionals =
+      ./unpackOptionals sha256:66c3e6f6f81418cf99342e1dba739617c01af4b27c1ca5e2e1d7bce64a522e22
+    ? ./unpackOptionals
```
Comment (german1608) [inline Prelude/Map/package.dhall:24]:
```suggestion
      ./unpackOptionals.dhall sha256:66c3e6f6f81418cf99342e1dba739617c01af4b27c1ca5e2e1d7bce64a522e22
    ? ./unpackOptionals.dhall
```

File: Prelude/Map/unpackOptionals.dhall [hunk 2]
```diff
@@ -0,0 +1,38 @@
+-- | Unpack Optionals from a Map, omitting None values.
+
+let List/concatMap =
+        ../List/concatMap sha256:3b2167061d11fda1e4f6de0522cbe83e0d5ac4ef5ddf6bb0b2064470c5d3fb64
+      ? ../List/concatMap
```
Comment (german1608) [inline Prelude/Map/unpackOptionals.dhall:5]:
```suggestion
        ../List/concatMap.dhall sha256:3b2167061d11fda1e4f6de0522cbe83e0d5ac4ef5ddf6bb0b2064470c5d3fb64
      ? ../List/concatMap.dhall
```

File: Prelude/Map/unpackOptionals.dhall [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+
+let Map/Entry =
+        ./Entry sha256:f334283bdd9cd88e6ea510ca914bc221fc2dab5fb424d24514b2e0df600d5346
+      ? ./Entry
```
Comment (german1608) [inline Prelude/Map/unpackOptionals.dhall:9]:
```suggestion
        ./Entry.dhall sha256:f334283bdd9cd88e6ea510ca914bc221fc2dab5fb424d24514b2e0df600d5346
      ? ./Entry.dhall
```

File: Prelude/Map/unpackOptionals.dhall [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+let Map/Type =
+        ./Type sha256:210c7a9eba71efbb0f7a66b3dcf8b9d3976ffc2bc0e907aadfb6aa29c333e8ed
+      ? ./Type
```
Comment (german1608) [inline Prelude/Map/unpackOptionals.dhall:13]:
```suggestion
        ./Type.dhall sha256:210c7a9eba71efbb0f7a66b3dcf8b9d3976ffc2bc0e907aadfb6aa29c333e8ed
      ? ./Type.dhall
```

File: Prelude/Map/unpackOptionals.dhall [hunk 5]
```diff
@@ -0,0 +1,38 @@
+-- | Unpack Optionals from a Map, omitting None values.
```
Comment (german1608) [inline Prelude/Map/unpackOptionals.dhall:1]: What about:
```suggestion
-- | Unpack Optionals from a `Map`, omitting `None` values.
```

**Comment by JackKelly-Bellroy** [comment]
Thanks. I applied your changes; let me know if they should be squashed into a single commit.

`./scripts/lint-prelude.sh` downloaded 850 MiB of stuff and charged off to rebuild low-level tools like `bison`, `gnum4` and `binutils`, so I killed it.

File: Prelude/Map/unpackOptionals.dhall [hunk 6]
```diff
@@ -0,0 +1,38 @@
+-- | Unpack Optionals from a `Map`, omitting `None` values.
```
Comment (sjakobi) [inline Prelude/Map/unpackOptionals.dhall:1]: That seems pretty vague. For example, the description should say that the `Optional`s must be in the values, not the keys.

File: Prelude/Map/unpackOptionals [hunk 7]
```diff
@@ -0,0 +1,2 @@
+  ./unpackOptionals.dhall sha256:66c3e6f6f81418cf99342e1dba739617c01af4b27c1ca5e2e1d7bce64a522e22
+? ./unpackOptionals.dhall
```
Comment (sjakobi) [inline Prelude/Map/unpackOptionals:2]: Since we don't have a backward compatibility constraint here, I think this file is unnecessary.

Comment (JackKelly-Bellroy) [reply Prelude/Map/unpackOptionals]: Thanks, removed.

File: Prelude/Map/unpackOptionals.dhall (hunk 6 above)
Comment (JackKelly-Bellroy) [reply Prelude/Map/unpackOptionals.dhall]: Rewritten, hopefully it's clearer.

File: Prelude/Map/unpackOptionals.dhall [hunk 8]
```diff
@@ -0,0 +1,39 @@
+-- | Turn a `Map k (Optional v)` into a `Map k v` by dropping all
+-- entries with value `None`.
```
Comment (sjakobi) [inline Prelude/Map/unpackOptionals.dhall:2]: @german1608 Is this correct?

```suggestion
--| Turn a `Map k (Optional v)` into a `Map k v` by dropping all
--  entries with value `None`.
```

(We should integrate  `dhall-docs` into CI.)

Comment (german1608) [reply Prelude/Map/unpackOptionals.dhall:2]: Yes, that is correct. Well spotted!

**Comment by JackKelly-Bellroy** [comment]
Excellent. When can we ship this?

**Comment by Gabriel439** [comment]
@JackKelly-Bellroy: There will be another release coming up soon.  By default we release on a roughly monthly cadence and the last release was about a month ago

**Comment by JackKelly-Bellroy** [comment]
Thanks for the heads-up @Gabriel439 . Looking forward to it.

**Comment by Gabriel439** [comment]
@JackKelly-Bellroy: The pull request to cut the next release is up here: https://github.com/dhall-lang/dhall-lang/pull/1059

**Pull request merged**